src/ArbitrumFoundationVestingWallet.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts-upgradeable/finance/VestingWalletUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {IERC20VotesUpgradeable} from "./Util.sol";
+
+interface IL2ArbitrumGoverner {
+    function token() external view returns (IERC20VotesUpgradeable);
+    function EXCLUDE_ADDRESS() external view returns (address);
+}
+
+/**
+ * @notice A wallet for foundation owned founds as per AIP-1.1 specification.
+ * DAO can migrate funds to new wallet.
+ * Wallet vests funds over time on a linear schedule.
+ * Governance votes are delegated to exclude address.
+ * Beneficiary can be updated by DAO.
+ */
+
+contract ArbitrumFoundationVestingWallet is VestingWalletUpgradeable, OwnableUpgradeable {
+    using SafeERC20 for IERC20;
+
+    address private _beneficiary;
+
+    event NewBeneficiary(address newBeneficiary);
+    event TokenMigrated(address token, uint256 amount, address destination);
+    event EthMigrated(uint256 amount, address destination);
+
+    constructor() {
+        _disableInitializers();
+    }
+    /// @param _beneficiaryAddress Can release funds and receives released funds
+    /// @param _startTimestamp The time to start vesting
+    /// @param _durationSeconds The time period for funds to fully vest
+    /// @param _arbitrumGoverner Core DAO Governer address
+    /// @param _owner Arbitrum DAO. Can migrate funds to new wallet and change beneficiary
+
+    function initialize(
+        address _beneficiaryAddress,
+        uint64 _startTimestamp,
+        uint64 _durationSeconds,
+        address _arbitrumGoverner,
+        address _owner
+    ) public initializer {
+        require(
+            _beneficiaryAddress != address(0),
+            "ArbitrumFoundationVestingWallet: zero beneficiary address"
+        );
+        require(
+            _arbitrumGoverner != address(0),
+            "ArbitrumFoundationVestingWallet: zero arbitrumGoverner address"
+        );
+
+        // init vesting wallet
+        // first argument (beneficiary) is unused by contract; a dummy value is provided
+        __VestingWallet_init(address(1), _startTimestamp, _durationSeconds);
+        _beneficiary = _beneficiaryAddress;
```

Comment:
should we manually emit the NewBeneficiary event here?
another option is having an internal _setBeneficiary function that we call that emits the event